query/iterator.gen.go.tmpl

```diff
@@ -409,14 +415,44 @@ func (h *{{$k.name}}SortedMergeHeap) Less(i, j int) bool {
 		} else if xTags, yTags := x.Tags.Subset(h.opt.Dimensions), y.Tags.Subset(h.opt.Dimensions); !xTags.Equals(&yTags) {
 			return xTags.ID() < yTags.ID()
 		}
+
+		if x.Time == y.Time {
+			if len(x.Aux) > 0 && len(x.Aux) == len(y.Aux) {
+				for i := 0; i < len(x.Aux); i++ {
+					v1, ok1 := x.Aux[i].(string)
+					v2, ok2 := y.Aux[i].(string)
+					if !ok1 || !ok2 {
+						// Unsupported types used in Aux fields. Maybe they
+						// need to be added here?
+						break
+					}
+					return v1 < v2
+				}
+			}
+		}
 		return x.Time < y.Time
 	}
```

Comment:
`strings.Compare` is always slower than using the builtin operator so I didn't move the comparison into a separate function because it becomes clunky when you want to do either `a < b` or `a > b`. 

A little bit of duplication seemed appropriate in this case.